osu.Game/Database/LegacyBeatmapExporter.cs

```diff
@@ -67,7 +67,12 @@ public LegacyBeatmapExporter(Storage storage)
 
                 hitObject.StartTime = Math.Floor(hitObject.StartTime);
 
-                if (hitObject is not IHasPath hasPath || BezierConverter.CountSegments(hasPath.Path.ControlPoints) <= 1) continue;
+                if (hitObject is not IHasPath hasPath) continue;
+
+                int segmentCount = BezierConverter.CountSegments(hasPath.Path.ControlPoints);
+
+                if (segmentCount == 0 || (segmentCount == 1 && hasPath.Path.ControlPoints[0].Type != PathType.BSpline))
+                    continue;
```

Comment:
I'm pretty sure this should be something like

```suggestion
                if (segmentCount <= 1 && hasPath.Path.ControlPoints.All(p => p.Type != PathType.BSpline))
```

because you can have multiple B-spline segments in a slider. And as soon as you get one you need to convert (but you also need to convert if you have _more_, or if it isn't _first_, which the current condition isn't checking).